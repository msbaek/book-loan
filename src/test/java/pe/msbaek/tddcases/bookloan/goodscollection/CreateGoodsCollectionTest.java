package pe.msbaek.tddcases.bookloan.goodscollection;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.List;

class CreateGoodsCollectionTest {
    private CreateGoodsCollection createGoodsCollection;
    private InMemoryGoodsCollectionRepository repository;

    @BeforeEach
    void setUp() {
        repository = new InMemoryGoodsCollectionRepository();
        createGoodsCollection = new CreateGoodsCollection(repository);
    }

    @Test
    void createGoodsCollection() {
        CreateGoodsCollection.CreateGoodsCollectionRequest request = new CreateGoodsCollection.CreateGoodsCollectionRequest("상품군 이름", List.of("GD00111839", "GD00111838", "GD00111836"));
        Long id = createGoodsCollection.createGoodsCollection(request);
        GoodsCollection result = repository.findById(id).get();
        Approvals.verify(result);
    }
}