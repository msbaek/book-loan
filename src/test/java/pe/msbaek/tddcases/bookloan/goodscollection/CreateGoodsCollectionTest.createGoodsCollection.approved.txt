---
id: 12
name: "상품군 이름"
createdBy: 1
createdAt: "2003-05-03T10:11:12.134567"
goodsCollectionItems:
- id: 34
  goodsNo: 112218
  goodsId: "GD00112218"
  barcode: "8804775369223"
- id: 35
  goodsNo: 112216
  goodsId: "GD00112216"
  barcode: "9000000112216"
- id: 36
  goodsNo: 112215
  goodsId: "GD00112215"
  barcode: "9000000112215"
